### PR TITLE
IEP-42 product information log

### PR DESCRIPTION
**Pull request by sigmaaa**
Was added a product information log which includes the following information:

1. IDF Plugins Release version
2. Operating system
3. Java Version
4. Eclipse Core version
5. Eclipse CDT version
6. idf_tools.py tools version info
7. Python and CMake version
8. Idf_tools.py export path output

To see product information log, user must click on "Product information" help menu item:
IEP-42

The Output looks like this:

IEP-42 product info log

ProductInformationHandler extends ListInstalledToolsHandler, so If some tools are not installed or IDF_PATH is not set, the user will see information dialog like this: 
IEP-308

### CODE REVIEW

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	{
+		super.execute(event);
+		console.println(Messages.JavaRuntimeVersionMsg + System.getProperty("java.runtime.version"));
+		console.println(Messages.OperationSystemMsg + System.getProperty("os.name").toLowerCase());
+		console.println(Messages.EclipseCDTMsg + Platform.getBundle("org.eclipse.cdt").getVersion().toString());
+		console.println(Messages.PythonIdfEnvMsg + getPythonExeVersion(IDFUtil.getIDFPythonEnvPath()));
+		console.println(Messages.PythonPathMsg + getPythonExeVersion("python"));
+		showEclipseVersion();
+		showEspIdfVersion();
+		return null;
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java:39]: How about re-ordering these messages

Operating System:
Java Version:
Eclipse Version:
Eclipse CDT Version:
IDF Eclipse Plugin version:
ESP-IDF version:
Python versions

Tools information:


Eclipse CDT Build environment variables
IDF_PATH
IDF_PYTHON_ENV_PATH
PATH

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+public class ProductInformationHandler extends ListInstalledToolsHandler
+{
+	@Override
+	public Object execute(ExecutionEvent event) throws ExecutionException
+	{
+		super.execute(event);
+		console.println(Messages.JavaRuntimeVersionMsg + System.getProperty("java.runtime.version"));
+		console.println(Messages.OperationSystemMsg + System.getProperty("os.name").toLowerCase());
+		console.println(Messages.EclipseCDTMsg + Platform.getBundle("org.eclipse.cdt").getVersion().toString());
+		console.println(Messages.PythonIdfEnvMsg + getPythonExeVersion(IDFUtil.getIDFPythonEnvPath()));
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java:35]: add non-nls

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	private void showEclipseVersion()
+	{
+		Wini ini;
+		try
+		{
+			ini = new Wini(new File(Platform.getInstallLocation().getURL().getPath() + "configuration/config.ini"));
+			console.println(Messages.EclipseMsg + ini.get("?", "eclipse.buildId"));
+		}
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java:91]: See if you can use this
`Version version = Platform.getBundle("org.eclipse.platform").getVersion();`

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/messages.properties [hunk 4]
```diff
@@ -26,4 +26,10 @@ InstallToolsHandler_InstallingToolsMsg=Installing tools...
 InstallToolsHandler_ItWilltakeTimeMsg=This can take a while. Please be patient.
 InstallToolsHandler_ToolsCompleted=Install tools completed.
 ListInstalledTools_MessageTitle=Message
-ListInstalledTools_MissingIdfPathMsg=Some tools are not installed or IDF_PATH is not set. To install them go to Help -> ESP-IDF Tools Manager -> Install Tools. Do you want to install the missing tools now?
\ No newline at end of file
+ListInstalledTools_MissingIdfPathMsg=Some tools are not installed or IDF_PATH is not set. To install them go to Help -> ESP-IDF Tools Manager -> Install Tools. Do you want to install the missing tools now?
+JavaRuntimeVersionMsg=Java runtime version: 
+OperationSystemMsg=Operation system:
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/messages.properties:31]: typo - "Operating System"

**Review by kolipakakondal — CHANGES_REQUESTED** [review]

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
+package com.espressif.idf.ui.help;
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java:6]: https://github.com/espressif/idf-eclipse-plugin/blob/cdaf7a1e2089498ec9e7f874cb037a9981f8ee60/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java#L1-L6

@kolipakakondal class copy-right info was added

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java (hunk 1 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java:48]: https://github.com/espressif/idf-eclipse-plugin/blob/cdaf7a1e2089498ec9e7f874cb037a9981f8ee60/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java#L32-L46

@kolipakakondal messages were reordered, but Tools information is going first. The reason for that, 
super.execute(event); activates and clears console output. 

So the output is next:
Tools information:

Eclipse CDT Build environment variables: (Read all environment variables defined and print)
IDF_PATH
IDF_PYTHON_ENV_PATH
PATH

Operating System:
Java Version:
Eclipse Version:
Eclipse CDT Version:
IDF Eclipse Plugin version:
ESP-IDF version:
Python versions

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java (hunk 2 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java]: https://github.com/espressif/idf-eclipse-plugin/blob/cdaf7a1e2089498ec9e7f874cb037a9981f8ee60/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java#L32-L46
@kolipakakondal non-nls was added.

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java (hunk 3 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java]: https://github.com/espressif/idf-eclipse-plugin/blob/cdaf7a1e2089498ec9e7f874cb037a9981f8ee60/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/help/ProductInformationHandler.java#L42
@kolipakakondal it works.Thank you for pointing that out

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/messages.properties (hunk 4 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/messages.properties]: https://github.com/espressif/idf-eclipse-plugin/blob/cdaf7a1e2089498ec9e7f874cb037a9981f8ee60/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/update/messages.properties#L31
@kolipakakondal fixed that. Thx

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Pull request merged**